docs/src/architecture.md

```diff
@@ -1,30 +1,166 @@
+multiple locations, preferably in a continuous and instantaneous way, by adopting
+either of the following two strategies:
+
+* *storage-level replication*, normally persistent volumes
+* *application-level replication*, in this specific case PostgreSQL
+
+CloudNativePG relies on application-level replication, for a simple reason: the
+PostgreSQL database management system comes with robust and reliable
+built-in **physical replication** capabilities based on **Write Ahead Log (WAL)
+shipping**, and used in production by millions of users all over the world for
+over a decade.
 
 PostgreSQL supports both asynchronous and synchronous streaming replication
 over the network, as well as asynchronous file-based log shipping (normally
 used as a fallback option, for example, to store WAL files in an object store).
 Replicas are usually called *standby servers* and can also be used for
 read-only workloads, thanks to the *Hot Standby* feature.
 
+!!! Important
+    **We recommend against storage-level replication with PostgreSQL**, although
+    CloudNativePG allows you to adopt that strategy. For more information, please refer
+    to the talk given by Chris Milsted and Gabriele Bartolini at KubeCon NA 2022 entitled
+    ["Data On Kubernetes, Deploying And Running PostgreSQL And Patterns For Databases In a Kubernetes Cluster"](https://www.youtube.com/watch?v=99uSJXkKpeI&ab_channel=CNCF%5BCloudNativeComputingFoundation%5D)
+    where this topic was covered in an exhaustive manner.
+
+
+## Kubernetes architecture
+
+Kubernetes natively provides the possibility to span across separate physical
+locations - also known as data centers, failure zones, or more frequently
+**availability zones** - connected to each other via redundant, low-latency,
+private network connectivity.
+
+Being a distributed system, the recommended minimum number of availability
+zones for a Kubernetes cluster is three (3), providing a way for the control
+plane to be resilient over the failure of a single zone.
+For details, please refer to
+["Running in multiple zones"](https://kubernetes.io/docs/setup/best-practices/multiple-zones/).
+This means that **each data center is active at any time** and can run workloads
+simultaneously.
+
+!!! Note
+    Most of the public Cloud Providers' managed Kubernetes services already
+    provide 3 or more availability zones in each region.
+
+### Multi-availability zone Kubernetes clusters
+
+The multi-availability zone Kubernetes architecture, having three (3) or more
+zones, is the one that we recommend for PostgreSQL usage.
+This scenario is typical of Kubernetes services managed by Cloud Providers.
+
+![Kubernetes cluster spanning over 3 independent data centers](./images/k8s-architecture-3-az.png)
+
+Such an architecture enables the CloudNativePG operator to control the full
+lifecycle of a `Cluster` resource within a single Kubernetes cluster, by
+treating all the availability zones as active: this includes, among the others,
```

Comment:
want to double-check what we want to convey with "zones as active"

Reply:
It is primarily from an "application point of view". With the active/passive scenario that some large Kubernetes providers suggest when facing a 2 data centre infrastructure, only one Kubernetes cluster can be considered active at any time - even though the other one could be for example replicating data using Postgres replica clusters to "pre-heat" the database cluster having the designated primary ready to be promoted. However, applications are redirected only to the first Kubernetes cluster.

I am sharing some links here:

- https://cloud.redhat.com/blog/a-guide-to-high-availability/disaster-recovery-for-applications-on-openshift
- https://www.ibm.com/cloud/architecture/articles/hadr-containers/hadr-containers-3-two-clusters-active-active-passive/

Thoughts?

Reply:
makes sense. I've been adding a bit more verbosity to clarify some of the more technical wording.